### PR TITLE
Enable dynamic model selection via API call

### PR DESCRIPTION
**Pull request by HabebNawatha**
# What does this PR do?
This pull request introduces an API-driven approach to dynamically fetch available models from the server, allowing users to select a model at runtime rather than relying on a hardcoded model. The fetched models are listed, and users can choose the model they want to use, improving flexibility and adaptability of the system.

## Sources

Please link relevant resources if necessary.

## Feature/Issue validation/testing/test plan

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration or test plan.

- [X] Test A : Fetch available models when the server is online.
Logs for Test A : Successfully retrieved models, user prompt shows available models, user can select any model dynamically.

- [X] Test B : Handle no models available scenario.
Logs for Test B :  If no models are available, the system displays a clear error message: "No models available in the distribution endpoint"

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/meta-llama/llama-stack-apps/blob/main/CONTRIBUTING.md#pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

Thanks for contributing 🎉!

### CODE REVIEW

File: examples/inference/client.py [hunk 1]
```diff
@@ -18,14 +18,36 @@ async def run_main(host: str, port: int, stream: bool = True):
     client = LlamaStackClient(
         base_url=f"http://{host}:{port}",
     )
+    try:
```
Comment (ashwinb) [inline examples/inference/client.py:21]: you can avoid an explicit `try-catch` and just let the natural exception bubble up

File: examples/inference/client.py [hunk 2]
```diff
@@ -18,14 +18,36 @@ async def run_main(host: str, port: int, stream: bool = True):
...
+        models_response = client.models.list()  # Fetch models list from server
+        available_models = [model.identifier for model in models_response]
+        if not available_models:
+            print("No available models in the distribution endpoint.")
```
Comment (ashwinb) [inline examples/inference/client.py:25]: probably do a `cprint(..., "red")`

File: examples/inference/client.py [hunk 3]
```diff
@@ -18,14 +18,36 @@ async def run_main(host: str, port: int, stream: bool = True):
...
+    print("Available Models:")
+    for idx, model_name in enumerate(available_models, start=1):
+        print(f"{idx}. {model_name}")
+    model_input = input(f"Select a model by number (or press Enter for default - {available_models[0]}): ")
+
+    try:
+        model_index = int(model_input) - 1 if model_input else 0
+        selected_model = available_models[model_index]
+    except (ValueError, IndexError):
+        print("Invalid selection. Using default model.")
```
Comment (ashwinb) [inline examples/inference/client.py:40]: you should do a loop here -- don't accept input until it is valid

**Review by ashwinb — CHANGES_REQUESTED** [review]
thanks, a few comments inline

File: examples/inference/client.py (hunk 1 above)
Comment (HabebNawatha) [reply examples/inference/client.py]: @ashwinb Thank you for your review! pushed the new changes!

File: examples/inference/client.py (hunk 2 above)
Comment (HabebNawatha) [reply examples/inference/client.py]: @ashwinb Thank you for your review! pushed the new changes!

File: examples/inference/client.py (hunk 3 above)
Comment (HabebNawatha) [reply examples/inference/client.py]: @ashwinb Thank you for your review! pushed the new changes!

**Review by ashwinb — APPROVED** [review]

**Pull request merged**